### PR TITLE
feat: realtime user analytics

### PR DESCRIPTION
**Pull request by vsrivatsa-juspay**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This PR introduces a real-time analytics feature that allows merchants to track distinct payment IDs on their dashboards. The key components of this feature include a script to create a ClickHouse table with a 1-minute Time-To-Live (TTL) for the data, and an analytics service that counts distinct payment IDs.
Also add the the http server for vector to collect and push logs to kafka in this PR.

## Changes:

**ClickHouse Table Creation Script:**
Added a script to create a ClickHouse table with a 1-minute TTL. This table will store payment events and automatically purge entries older than 1 minute, ensuring that only recent data is kept for real-time analytics.

**Analytics Service:**
Implemented an analytics service that queries the ClickHouse table to count distinct payment IDs. This provides real-time insights into the number of unique payments processed within the last minute.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

**1. Start the necessary services using Docker Compose**
```
docker compose up -d kafka-ui clickhouse-server
```
**2. Load the SDK and simulate payments**
**3. Verify Real-Time Analytics**
```
curl --location 'localhost:8080/analytics/v1/metrics/active_payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'Authorization: Bearer {token}' \
--data '[
    {
        "timeRange": {
            "startTime": "2024-06-24T18:30:00Z",
            "endTime": "2024-06-24T18:31:00Z"
        },
        "metrics": [
            "active_payments"
        ]
    }
]'
```
```
{
    "queryData": [
        {
            "active_payments": 2
        }
    ],
    "metaData": [
        {
            "current_time_range": {
              "start_time": "2024-06-24T18:30:00Z",
              "end_time": "2024-06-24T18:31:00Z"
            }
        }
    ]
}
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: config/vector.yaml [hunk 1]
```diff
@@ -132,3 +145,15 @@ sinks:
...
+    type: kafka
+    encoding:
+      codec: json
+      except_fields:
+        - "path"
+        - "source_type"
+    inputs:
+      - "sdk_transformed"
+    bootstrap_servers: kafka0:29092
+    topic: hyper-sdk-logs
```
Comment (lsampras) [inline config/vector.yaml:159]: can we use the `<merchant_id>_<payment_id>` as the key field?

Comment (lsampras) [inline config/vector.yaml:159]:
```suggestion
    topic: hyper-sdk-logs
    key_field: "{{ .merchant_id }}{{ .payment_id }}"
```

Comment (lsampras) [reply config/vector.yaml:159]: https://vector.dev/docs/reference/configuration/sinks/kafka/#key_field

**Review by lsampras — APPROVED** [review]

**Review by lsampras — APPROVED** [review]

**Review by ivor-juspay — APPROVED** [review]

File: config/vector.yaml (hunk 1 above)
Comment (vsrivatsa-juspay) [reply config/vector.yaml:159]: This is giving an error. Had to revert for now

**Review by lsampras — APPROVED** [review]

**Review by ivor-juspay — APPROVED** [review]

**Pull request merged**